extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -264,7 +265,7 @@ public UUID findFirst(JobAuthorization.State jobState) {
   }
 
   @Override
-  public <T extends DataModel> void update(UUID jobId, String key, T model) {
+  public <T extends Serializable> void update(UUID jobId, String key, T model) {
```

Comment:
didn't look too closely so not sure it's possible, but do you think it makes sense to do any refactoring/visibility changes to ensure operations always go thru the executor when intended, and not thru jobstore directly?

this may make more sense if you make an idempotent exporter as well - which doesn't matter for the end result of the transfer but could preserve quota/resources from exporting the same thing twice

Reply:
Do you mean to lock down job store so integratos can't use it?

Flickr still has a fairly valid use case for it I think, so we couldn't lock it down from everyone.

Reply:
yea like lock down jobstore so integrators are forced to go thru the executor/caching layer. would it make sense for flickr to use the executor directly?

Reply:
No their use case is on the export side so it wouldn't make sense to use the executor.

Reply:
oh did you see my other point about also doing an export executor?

the idea being that it could preserve quota/resources from exporting the same thing twice (though it doesn't affect the end result data like the importer does)

Reply:
oh I didn't, ya we should do one of those as well, but I think that is best left to a different PR as this one is already huge.  Flickr also has a case where they need to use the jobstore in the importer as well, and the executor is not a good fit for the use case.

Reply:
ack - in that case I think it's worth thinking about other ways to make it so integrators don't accidentally bypass the caching & double-import, but i'm not sure the best way. i'm fine with it as is for now, though.